core/src/main/java/org/mskcc/cbio/portal/scripts/ImportGenesetData.java

```diff
@@ -0,0 +1,324 @@
+
+
+public class ImportGenesetData extends ConsoleRunnable {
+    
+	public static int skippedGenes;
+    
+    @Override
+    public void run() {
+        try {
+            String progName = "ImportGenesetData";
+            String description = "Import gene set data files.";
+            // usage:     --data <data_file.txt> --supp <supp_file.txt>
+            // optional:  --new-version <version>
+            //            --update-info
+
+            OptionParser parser = new OptionParser();
+            OptionSpec<String> data = parser.accepts("data", "Geneset data file")
+                    .withRequiredArg().ofType(String.class);
+            OptionSpec<String> supp = parser.accepts("supp", "Option gene set supplemental data file")
+                    .withRequiredArg().ofType(String.class);
+            OptionSpec<String> newVersionArgument = parser.accepts("new-version", "New version of gene sets. This will REMOVE GSVA data and hierarchy.")
+            	.withRequiredArg().ofType(String.class);
+            parser.accepts("update-info", "Only update the info of gene sets (Ref Link, name, description)");
+
+            // For testing purposes, we don't want the prompt, so make extra argument.
+            parser.accepts("confirm-delete-all-genesets-hierarchy-genesetprofiles", "This will avoid the command prompt being shown.");
+
+            OptionSet options = null;
+            try {
+                options = parser.parse(args);
+            }
+            catch (Exception ex) {
+                throw new UsageException(
+                        progName, description, parser, 
+                        ex.getMessage());
+            }
+            
+            // if neither option was set then throw UsageException
+            if (!options.has(data) && !options.has(supp)) {
+                throw new UsageException(
+                        progName, description, parser,
+                        "'data' and/or 'supp' argument required");
+            }
+            
+            // Check options
+            boolean updateInfo = options.has("update-info");
+            boolean newVersion = options.has("new-version");
+            boolean confirmEmptyGenesets = options.has("confirm-delete-all-genesets-hierarchy-genesetprofiles");
+            
+            // Check current version
+            GenesetInfo genesetInfo = new GenesetInfo();
+            genesetInfo = DaoGenesetInfo.getGenesetInfo();
+            String databaseVersionValue = genesetInfo.getVersion();
+            
+            // Check new version
+            genesetInfo = new GenesetInfo();
+            genesetInfo.setVersion(options.valueOf(newVersionArgument));
+            
+            // Check if gene set tables are empty. In that case, no extra argument is required
+
+        	// Geneset tables are not empty. Need to specify if we update info or add new version
+            if ((!updateInfo && !newVersion) || (updateInfo && newVersion)) {      
```

Comment:
Correct! I've changed this to:
`        	// Required to specify if we update info or add new version`